### PR TITLE
Misc coordinate docstring fixes

### PR DESCRIPTION
**Pull request by dstansby**
### Description

I discovered https://pypi.org/project/velin/, which is a small command line utility to automatically fix small/common numpydoc issues in docstrings. I've run it on `astropy`, and it seemed to catch a few things. This PR is fixes to `coordinates`.

- I've checked all these by eye to make sure they look sensible
- I figured it would be best to open one PR/sub-module

Fixes #<Issue Number>

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by pllim — COMMENTED** [review]
It is good to have some clean-ups. Thanks! Minor comments below.

File: astropy/coordinates/angle_utilities.py [hunk 1]
```diff
@@ -106,9 +106,6 @@ def offset_by(lon, lat, posang, distance):
         The position of the final point.  If any of the angles are arrays,
         these will contain arrays following the appropriate `numpy` broadcasting rules.
         0 <= lon < 2pi.
-
-    Notes
```
[the remark this answers is not in the document]
Comment (dstansby) [reply astropy/coordinates/angle_utilities.py:110]: Nope, looks like it's just an unintentional heading as there's no Notes section in the current rendered docs: https://docs.astropy.org/en/latest/api/astropy.coordinates.offset_by.html

File: astropy/constants/utils.py [hunk 2]
```diff
@@ -46,19 +44,15 @@ def _set_c(codata, iaudata, module, not_in_module_only=True, doclines=None,
...
-
     not_in_module_only : bool
         If ``True``, constants that are already in the namespace
         of ``module`` will not be modified.
-
     doclines : list or None
         If a list is given, this list will be modified in-place to include
         documentation of modified constants. This can be used to update
         docstring of ``module``.
-
```
[the remark this answers is not in the document]
Comment (dstansby) [reply astropy/constants/utils.py:61]: 👍 - I'll remove the newline changes.

**Comment by pllim** [comment]
Failure is unrelated. I wonder if it is because of the new `pytest-astropy-header` release... 😱

File: astropy/coordinates/angle_utilities.py [hunk 3]
```diff
@@ -216,8 +213,6 @@ def uniform_spherical_random_volume(size=1, max_radius=1):
         The number of points to generate.
     max_radius : number, quantity-like, optional
         A dimensionless or unit-ful factor to scale the random distances.
-    rng : `numpy.random.Generator`, optional
```
Comment (nstarman) [inline astropy/coordinates/angle_utilities.py:219]: While it's correct that this function doesn't accept an `rng` argument, I think one should be added!
Definitely as a followup, so it's right to remove it here.

**Review by nstarman — COMMENTED** [review]
Thanks @dstansby. Everything looks good to me. 🤞 if you have the time to run it elsewhere! I wish the numpydoc tool in v1.2 could do auto-correct!

Ping me again when this PR is rebased. Hopefully CI should be passing now.

File: astropy/coordinates/angle_utilities.py (hunk 3 above)
Comment (pllim) [reply astropy/coordinates/angle_utilities.py:219]: Maybe @adrn can explain. He added this in #11628 .

Comment (adrn) [reply astropy/coordinates/angle_utilities.py:219]: Ah, that docstring is vestigial ... I had originally added the rng option but see discussion here: #11628 and #11687

Comment (nstarman) [reply astropy/coordinates/angle_utilities.py:219]: Sounds a worthwhile Astropy-wide goal.

Comment (adrn) [reply astropy/coordinates/angle_utilities.py:219]: (I have a branch where I started down this path -- I'll see what state it's in!)

File: astropy/coordinates/sky_coordinate.py [hunk 4]
```diff
@@ -417,9 +417,9 @@ def _apply(self, method, *args, **kwargs):
         method : str or callable
             If str, it is the name of a method that is applied to the internal
             ``components``. If callable, the function is applied.
-        args : tuple
+        *args : tuple
```
Comment (mhvk) [inline astropy/coordinates/sky_coordinate.py:420]: Like elsewhere, probably delete `: tuple` and `: dict` here.

File: astropy/coordinates/sky_coordinate.py [hunk 5]
```diff
@@ -1751,12 +1752,12 @@ def contained_by(self, wcs, image=None, **kwargs):
             is being checked against. If not given the naxis keywords will be used
             to determine if the coordinate falls within the wcs footprint.
         **kwargs :
```
Comment (mhvk) [inline astropy/coordinates/sky_coordinate.py:1754]: Remove the colon?

**Review by mhvk — APPROVED** [review]
Two small comments, but I'm going to approve as this is an obvious improvement and shouldn't linger!

File: astropy/coordinates/sky_coordinate.py (hunk 5 above)
Comment (eteq) [reply astropy/coordinates/sky_coordinate.py:1754]:
```suggestion
        **kwargs
```

**Review by eteq — APPROVED** [review]

**Comment by eteq** [comment]
I went ahead and applied @mhvk's suggestions, since they were really straightforward, and I'll just go ahead and merge this later today if/when the CI passes so it doesn't go too stale.

**Comment by eteq** [comment]
I don't understand why RTD keeps failing, but it doesn't seem related to this PR as I see it in other recent PRs.  To be safe I just did a `tox -e build_docs` and it built without problems, so I think this is good to go.

**Comment by pllim** [comment]
It was "concurrency limit". Maybe CoCo should consider paying them to up the limit; is that possible?

**Pull request merged**